### PR TITLE
Implement displaying of network requests data

### PR DESCRIPTION
**Pull request by RedMickey**
https://github.com/microsoft/vscode-react-native/issues/1418

### CODE REVIEW

File: src/extension/networkInspector/requestBodyFormatters/imageFormatter.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import { IFormatter } from "./requestBodyFormatter";
+import { Response } from "../networkMessageData";
+
+export class ImageFormatter implements IFormatter {
+    public formatResponse(response: Response, contentType: string): string | any | null {
```
Comment (etatanova) [inline src/extension/networkInspector/requestBodyFormatters/imageFormatter.ts:8]: It seems that there should be a return type `string | null`, `any`, or something else.

File: src/extension/networkInspector/views/inspectorConsoleView.ts [hunk 2]
```diff
@@ -1,10 +1,229 @@
...
+
+    private setupConsoleLogsColor(systemColorTheme: SystemColorTheme): void {
+        if (systemColorTheme === SystemColorTheme.Light) {
+            this.consoleLogsColor = this.consoleLogsColors.Blue;
+        } else {
+            this.consoleLogsColor = this.consoleLogsColors.Orange;
+        }
+    }
+
+    private handleRequest(data: Request) {
```
Comment (etatanova) [inline src/extension/networkInspector/views/inspectorConsoleView.ts:84]: Could you please add a return type `void` to all these kind of methods of this class?

File: src/extension/networkInspector/requestBodyFormatters/requestBodyFormatter.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+                    dataArr[i] = binStr.charCodeAt(i);
+                }
+                // The request is gzipped, so convert the base64 back to the raw bytes first,
+                // then inflate. pako will detect the BOM headers and return a proper utf-8 string right away
+                return pako.inflate(dataArr, { to: "string" });
+            } catch (e) {
+                // on iOS, the stream send to flipper is already inflated, so the content-encoding will not
+                // match the actual data anymore, and we should skip inflating.
+                // In that case, we intentionally fall-through
+                if (!("" + e).includes("incorrect header check")) {
```
Comment (etatanova) [inline src/extension/networkInspector/requestBodyFormatters/requestBodyFormatter.ts:85]: Maybe it's better to do type casting by using `as string`, `<string>`, or `.toString()` here?

File: src/extension/settingsHelper.ts [hunk 4]
```diff
@@ -131,6 +132,20 @@ export class SettingsHelper {
         return "";
     }
 
+    public static getNetworkInspectorConsoleLogsColorTheme(): SystemColorTheme {
+        const workspaceConfiguration = vscode.workspace.getConfiguration(
+            "react-native-tools.networkInspector",
+            null,
+        );
+        if (workspaceConfiguration.has("consoleLogsColorTheme")) {
+            let consoleLogsColorTheme: string = ConfigurationReader.readString(
```
Comment (etatanova) [inline src/extension/settingsHelper.ts:141]: It seems we may use `const` here
```suggestion
            const consoleLogsColorTheme: string = ConfigurationReader.readString(
```

**Review by JiglioNero — APPROVED** [review]

**Review by etatanova — APPROVED** [review]

**Pull request merged**